libs/js-sdk/src/client.ts

```diff
@@ -88,26 +91,48 @@ export function createBrowserClient({
     return {
         client: createBrowserFbClient(state),
         get initialised() {
-            return initialisedPromise.completed
+            return (
+                initialisedPromise.completed &&
+                updatedPromise.every((p) => p.completed)
+            )
         },
         waitForInitialised() {
             return new Promise((resolve) => {
                 const interval = setInterval(() => {
-                    if (initialisedPromise.completed) {
+                    if (this.initialised) {
                         clearInterval(interval)
                         resolve(true)
                     }
                 }, 100)
             })
         },
-        updateAudiences(updatedAudiences: string[]) {
+        initialisedChanged(callback) {
+            initialisedCallback = callback
```

Comment:
Is it worth making this an array of callbacks instead?

Also could be a good idea to return a function which unsubscribes (removes from the list)

Reply:
The though hit me as well after I had done this... Can't think of a reason but a developer might want to check if the initialisedChanged in other places as well...